core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java

```diff
@@ -1968,4 +1970,87 @@ public void dropNonEmptyNamespace() {
         .as("Namespace should not exist")
         .isFalse();
   }
+
+  @Test
+  public void registerView() {
+    C catalog = catalog();
+
+    // Register view is not yet supported for REST catalog
+    assumeThat(catalog instanceof RESTCatalog).isFalse();
+
+    TableIdentifier identifier = TableIdentifier.of("ns", "view");
+
+    if (requiresNamespaceCreate()) {
+      catalog.createNamespace(identifier.namespace());
+    }
+
+    View originalView =
+        catalog()
+            .buildView(identifier)
+            .withSchema(SCHEMA)
+            .withDefaultNamespace(identifier.namespace())
+            .withDefaultCatalog(catalog().name())
+            .withQuery("spark", "select * from ns.tbl")
+            .withProperty(GC_ENABLED, "false")
+            .create();
+
+    ViewOperations ops = ((BaseView) originalView).operations();
+    String metadataLocation = ops.current().metadataFileLocation();
+
+    assertThat(catalog.dropView(identifier)).isTrue();
+    assertThat(catalog.viewExists(identifier)).as("View must not exist").isFalse();
+
+    // view metadata should still exist after dropping the view as gc is disabled
+    assertThat(((BaseViewOperations) ops).io().newInputFile(metadataLocation).exists()).isTrue();
+
+    View registeredView = catalog.registerView(identifier, metadataLocation);
+
+    assertThat(registeredView).isNotNull();
+    assertThat(catalog.viewExists(identifier)).as("View must exist").isTrue();
+    assertThat(registeredView.schema().asStruct())
+        .as("Schema must match")
+        .isEqualTo(originalView.schema().asStruct());
+    assertThat(registeredView.currentVersion())
+        .as("Current version must match")
+        .isEqualTo(originalView.currentVersion());
```

Comment:
this should also be explicit and check similarly to how we check in `basicCreateView()` by additionally adding an `.isEqualTo(ImmutableViewVersion.builder()...)`